### PR TITLE
Upgrading to Bootstrap 4

### PR DESCRIPTION
**Pull request by jack-farley**
### Summary 

This pull request updates CU Reviews from bootstrap 3 to bootstrap 4. Currently, the client is largely updated, except for the class view page, which Andrew is updating and will add on top of this PR. 

- [x] Upgrades to bootstrap 4
- [x] Removes panels
- [x] Updates everything to use the new version of the grid system
- [x] some other minor changes

### Test Plan 

I have tested these updates on many screen sizes from a large monitor to a small window, but I haven't done too much testing on mobile yet.

### Linter Warnings 

Right now I am not seeing any linter warnings for the update. 

### Notes

If you are getting an error after pulling this branch, try removing your node modules and reinstalling.

### CODE REVIEW

File: client/src/ui/ClassView.jsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class ClassView extends Component {
       );
       return (
         <div>
-          <div className="hidden-lg hidden-xl hidden-md">
+          <div className=".d-none.d-xs-block.d-md-none">
```
[the remark this answers is not in the document]
Comment (jack-farley) [reply client/src/ui/ClassView.jsx:121]: Thanks, good catch. Wasn't working when I tested it, wasn't sure why - this seems to have fixed it though.

File: client/src/ui/PreviewCard.jsx [hunk 2]
```diff
@@ -186,13 +186,13 @@ export default class PreviewCard extends Component {
 
         {!this.props.transformGauges && (
           <div className="row gaugeHolder">
-            <div className="col-md-4 col-sm-4 col-xs-4 remove-left-padding">
+            <div className="col-lg-4 col-md-4 col-sm-4 col-xs-4 remove-left-padding">
```
Comment (kungpaogao) [inline client/src/ui/PreviewCard.jsx:189]:
```suggestion
            <div className="col-lg-4 col-md-4 col-sm-4 col-4 remove-left-padding">
```
there's no more `xs` breakpoint, so here and elsewhere `col-xs-{value}` should just be `col-{value}`

File: client/src/ui/UpdateReview.tsx [hunk 3]
```diff
@@ -90,33 +90,39 @@ export default class UpdateReview extends Component<Props, State> {
...
-              <div className="panel panel-default">
-                <div className="panel-body text-center">{review.difficulty}</div>
+              <div className="col-md-3 col-sm-6 col-xs-6">
+                <div className="card-body"> Overall Rating</div>
+                <div className="card-body"> Difficulty</div>
+                <div className="card-body"> Professor(s)</div>
               </div>
-              <div className="panel panel-default">
-                <div className="panel-body text-center">{review.professors}</div>
+              <div className="col-md-6 col-sm-12 col-xs-12">
```
Comment (kungpaogao) [inline client/src/ui/UpdateReview.tsx:118]: `col-xs` --> `col`

**Review by kungpaogao — CHANGES_REQUESTED** [review]
overall, looks great - looks like this was pretty tedious to do!

biggest thing I noticed (without checking out the branch locally) is that `col-xs` should be just `col` (https://getbootstrap.com/docs/4.0/migration/#grid-system)

**Review by AkashAryal — APPROVED** [review]
Looks Good! Thanks for the big PR!

**Review by kungpaogao — APPROVED** [review]
looks good :))

**Pull request merged**